fv3core/utils/corners.py

```diff
@@ -220,52 +221,116 @@ def definition(q: FloatField):
     stencil(q, origin=origin, domain=domain)
 
 
-def copy_sw_corner(q, direction, grid, kslice):
-    for j in range(grid.js - grid.halo, grid.js):
-        for i in range(grid.is_ - grid.halo, grid.is_):
-            if direction == "x":
-                q[i, j, kslice] = q[j, grid.is_ - i + 2, kslice]
-            if direction == "y":
-                q[i, j, kslice] = q[grid.js - j + 2, i, kslice]
+@gtscript.function
+def copy_corners_x(q):
+    from __externals__ import i_end, i_start, j_end, j_start
```

Comment:
why end before start?

Reply:
good catch! no idea! I'll change it

Reply:
Oh, looks like other gtscript functions in this file use that order, I probably copied it